x/claimsmanager/spec/README.md

```diff
@@ -129,11 +129,107 @@ type Proof struct {
 }
 ```
 
+### ProtocolData
+
+```protobuf
+// KeyedProtocolData is a keyed representation of protocoldata used in the
+// genesis file.
+message KeyedProtocolData {
+  string key = 1;
+  ProtocolData protocol_data = 3;
+}
+// ProtocolData is an arbitrary data type held against a given zone for the
+// determination of claim allocaton.
+message ProtocolData {
+  string type = 1;
+  bytes data = 2 [(gogoproto.casttype) = "encoding/json.RawMessage"];
+}
+// ProtocolDataType is an enumerated list of types of protocol data.
+enum ProtocolDataType {
+  option (gogoproto.goproto_enum_prefix) = false;
+  // Undefined action (per protobuf spec)
+  ProtocolDataTypeUndefined = 0;
+  ProtocolDataTypeConnection = 1;
+  ProtocolDataTypeOsmosisParams = 2;
+  ProtocolDataTypeLiquidToken = 3;
+  ProtocolDataTypeOsmosisPool = 4;
+  ProtocolDataTypeCrescentPool = 5 [deprecated = true];
+  ProtocolDataTypeSifchainPool = 6 [deprecated = true];
+  ProtocolDataTypeUmeeParams = 7;
+  ProtocolDataTypeUmeeReserves = 8;
+  ProtocolDataTypeUmeeInterestScalar = 9;
+  ProtocolDataTypeUmeeTotalBorrows = 10;
+  ProtocolDataTypeUmeeUTokenSupply = 11;
+  ProtocolDataTypeUmeeLeverageModuleBalance = 12;
+  ProtocolDataTypeCrescentParams = 13 [deprecated = true];
+  ProtocolDataTypeCrescentReserveAddressBalance = 14 [deprecated = true];
+  ProtocolDataTypeCrescentPoolCoinSupply = 15 [deprecated = true];
+  ProtocolDataTypeOsmosisCLPool = 16;
+}
+```
+
+## Events
+
+```protobuf
+message ClaimableEvent {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+  
+  string event_module = 0;              // module name doing the claim, “govbyproxy”, “participationrewards”, “airdrop” (types.ModuleName)
+  string event_name = 1;                // “epoch”, “proposal/cosmoshub-4/764”, “airdrop/cosmoshub-4”
+  Map<string, int64> heights = 2;       // heights across each chain for this event - when initialised create with the chainID and 0, for each relevant zone. For all zones, use all zones we have a connectionData for; airdrop proofs will require _latest_ (pool contents, etc.) to be submitted by the user.
+  max_claim_time time.Time = 3;         // when can we claim until? (Proposal end vote time, next epoch boundary, airdrop end)
+  // todo add more fields
+}
+```
+
 ## Messages
 
 Description of message types that trigger state transitions;
 
-`x/claimsmanager` does not alter state directly, it rather provides the mechanism for other modules to do so.
+### MsgSubmitClaim
+
+```protobuf
+// MsgSubmitClaim represents a message type for submitting a participation
+// claim regarding the given zone (chain).
+message MsgSubmitClaim {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+  string module_address = 1 [ json_name = "module_address", (cosmos_proto.scalar) = "cosmos.AddressString" ];
+  string event_name = 2 [ json_name = "zone" ];
+  // todo add more fields
+}
+```
+
+### MsgDeleteClaimableEvent
```

Comment:
Do we require this?

I think just having a method on the Keeper to do this is sufficient. It should be called by whichever module created the event in the first place. Should not require a tx to remove anything.

Reply:
Makes sense. I will remove it then. 

My idea about the design is to be controlled by contracts as well.

Reply:
That is a fair point. I’m not terribly familiar with how bindings between cw and go logic work.

Reply:
Contracts need proto messages that they can call. 
While writing rust contracts we need a QuickSilver proto lib that gives rust code for calling those endpoints.